state/machine.go

```diff
@@ -2230,6 +2273,24 @@ func removeUpgradeSeriesLockTxnOps(machineDocId string) []txn.Op {
 	}
 }
 
+func setUpgradeSeriesTxnOps(machineDocId string, unitIndex int, status params.UnitSeriesUpgradeStatus) []txn.Op {
+	unitField := fmt.Sprintf("prepareunits.%d.status", unitIndex)
+	return []txn.Op{
+		{
+			C:      machinesC,
+			Id:     machineDocId,
+			Assert: isAliveDoc,
+		},
+		{
+			C:      machineUpgradeSeriesLocksC,
```

Comment:
would we want to take the unit id and assert that 
prepareunits.%d.id
is what we expect?
making a change to the "nth" entry in an array always seems a bit concerning without it. I realize the array shouldn't be changing order, but certainly I would think we want to associate the status to the Id rather than the offset.